string language
string gender
string text
---
string errorMsg
string filepath